mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -1999,6 +2039,8 @@ OpFoldResult ExtractOp::fold(FoldAdaptor) {
     return val;
   if (auto val = foldScalarExtractFromFromElements(*this))
     return val;
+  SmallVector<Value> operands = {getVector()};
+  foldConstantOp(*this, operands);
   return OpFoldResult();
```

Comment:
This looks wrong. Even if you change the operation, you are still returning nullptr which means the folding didn't happen. Maybe make foldConstantOp return LogicalResult and check that.